pkg/stdlib/pkgs/task.cue

```diff
@@ -44,38 +56,48 @@ _EOF_
 	script: string
 	envs: [...{name: string, value: string}]
 	workspaceMounts: [...{workspace: #workspace, mountPath: string}]
+	secretMounts: [...{secret: #secret, mountPath: string}]
 }
 
 #workspace: {
 	name: string
 }
 
+#secret: {
+    name: string
+    items: [...{key: string, path: string}]
+}
+
 #PodTask: {
-	_scripts: string
-	_workspace: {...}
-	_volumes: [ for x in _workspace {name: x.name, emptyDir: {}}]
+	scripts_: string
+	workspaces_: {...}
+	secrets_: {...}
+	volumes_: [ for x in workspaces_ {name: x.name, emptyDir: {}}]
+	secretVolumes_: [ for x in secrets_ {name: x.name, secret: {secretName: x.name, items: x.items}}]
+	toolImage_: string
+	baseImage_: string
```

Comment:
It is recommended to put these in the section whose name starts with an underscore. for example
```
#PodTask: {
      _settings: {
          scripts_: string
          workspaces_: {...}
          secrets_: {...}
	  volumes_: [ for x in workspaces_ {name: x.name, emptyDir: {}}]
	  secretVolumes_: [ for x in secrets_ {name: x.name, secret: {secretName: x.name, items: x.items}}]
	  toolImage_: string
	  baseImage_: string
      }
}
```

Reply:
ok, I will make the changes. but it will reproduce the error "field "_xxx" not allowed in closed struct", if use "_setting" instead of "setting_". It may be caused by following codes:

val.FillObject(field.Value, field.Name), this func in following package can't support the field with name prefix "_" (the name starting with "_").

pkg/cue/model/value/value.go 

// StepByFields process the fields in order
func (val *Value) StepByFields(handle func(name string, in *Value) (bool, error)) error {
	for i := 0; ; i++ {
		field, end, err := val.fieldIndex(i)
		if err != nil {
			return err
		}

		if end {
			return nil
		}
		stop, err := handle(field.Name, field.Value)
		if err != nil {
			return errors.WithMessagef(err, "step %s", field.Name)
		}
		if stop {
			return nil
		}

		if !isDef(field.Name) {
			if err := val.FillObject(field.Value, field.Name); err != nil {  //throw error here
				return err
			}
		}

		if end {
			break
		}
	}
	return nil
}

Reply:
i made the changes, and committed again, please help review. thanks

Reply:
> ok, I will make the changes. but it will reproduce the error "field _xxx not allowed in closed struct", if use "_setting" instead of "setting_". It may be caused by following codes:
> 
> val.FillObject(field.Value, field.Name), this func in following package can't support the field with name prefix "_" (the name starting with "_").
> 
> pkg/cue/model/value/value.go
> 
> ```
> // StepByFields process the fields in order
> func (val *Value) StepByFields(handle func(name string, in *Value) (bool, error)) error {
> 	for i := 0; ; i++ {
> 		field, end, err := val.fieldIndex(i)
> 		if err != nil {
> 			return err
> 		}
> 
> 		if end {
> 			return nil
> 		}
> 		stop, err := handle(field.Name, field.Value)
> 		if err != nil {
> 			return errors.WithMessagef(err, "step %s", field.Name)
> 		}
> 		if stop {
> 			return nil
> 		}
> 
> 		if !isDef(field.Name) {
> 			if err := val.FillObject(field.Value, field.Name); err != nil {  //throw error here
> 				return err
> 			}
> 		}
> 
> 		if end {
> 			break
> 		}
> 	}
> 	return nil
> }
> ```

I see, this is a bug of `cue` 0.2.2-version.  Replace `_settings` with `#settings`, and please try again

Reply:
it is ok now. That was my fault. The func StepByFields will not be executed, if we don't put "_xxx" field under "steps" types (op.#Task). So it is ok that we put "_settings" field under #PodTask and #StepContainer types. I have submitted the codes. Please help review.